### PR TITLE
Web App & Function App Deploy - Review comments

### PR DESCRIPTION
**Pull request by vincent1173**

### CODE REVIEW

File: function-app-deploy/entrypoint.sh [hunk 1]
```diff
@@ -2,80 +2,81 @@
 
 set -e
 
-LINUX_APP_SUBSTRING='linux'
-
-if [[ -z $WEB_APP_NAME ]];
+if [[ -z $AZURE_APP_NAME ]];
 then 
-    echo "Required Web App Name. Provide value in WEB_APP_NAME variable" >&2
+    echo "Required Web App Name. Provide value in AZURE_APP_NAME variable" >&2
```
Comment (SumiranAgg) [inline function-app-deploy/entrypoint.sh]: Azure function name?

File: function-app-deploy/entrypoint.sh [hunk 2]
```diff
@@ -2,80 +2,81 @@
...
+        az webapp config appsettings set -g "${RESOURCE_GROUP_NAME}" -n "${AZURE_APP_NAME}" --settings WEBSITE_RUN_FROM_PACKAGE=1 > /dev/null
         sleep 10 # TODO: find whether this app setting is updated in Kudu
         echo "Set WEBSITE_RUN_FROM_PACKAGE = 1 successfully!"
     fi
-else
-    WEBSITES_ENABLE_APP_SERVICE_STORAGE=`az webapp config appsettings list -n ${WEB_APP_NAME} -g ${RESOURCE_GROUP_NAME} --query "[?(@.name=='WEBSITES_ENABLE_APP_SERVICE_STORAGE')].value" -o tsv`
-    if [[ ! $WEBSITES_ENABLE_APP_SERVICE_STORAGE == "true" ]];
-    then
-        echo "Setting App Setting WEBSITES_ENABLE_APP_SERVICE_STORAGE = true ..."
-        az webapp config appsettings set -g "${RESOURCE_GROUP_NAME}" -n "${WEB_APP_NAME}" --settings WEBSITES_ENABLE_APP_SERVICE_STORAGE=true > /dev/null
```
Comment (SumiranAgg) [inline function-app-deploy/entrypoint.sh]: missed out storage variable

File: function-app-deploy/entrypoint.sh (hunk 1 above)
Comment (vincentdass) [reply function-app-deploy/entrypoint.sh]: updated text.

File: function-app-deploy/entrypoint.sh (hunk 2 above)
Comment (vincentdass) [reply function-app-deploy/entrypoint.sh]: thanks, added again

File: function-app-deploy/entrypoint.sh [hunk 3]
```diff
@@ -2,80 +2,90 @@
...
 fi
-echo "Web App Name: ${WEB_APP_NAME}"
 
-RESOURCE_GROUP_NAME=`az resource list -n "${WEB_APP_NAME}" --resource-type "Microsoft.Web/Sites" --query '[0].resourceGroup' | xargs`
+RESOURCE_GROUP_NAME=`az resource list -n "${AZURE_APP_NAME}" --resource-type "Microsoft.Web/Sites" --query '[0].resourceGroup' | xargs`
+
 if [[ -z $RESOURCE_GROUP_NAME ]];
 then
-    echo "Web App '${WEB_APP_NAME}' should exist before deployment." >&2
+    echo "Web App '${AZURE_APP_NAME}' should exist before deployment." >&2
```
Comment (SumiranAgg) [inline function-app-deploy/entrypoint.sh]: function app

File: function-app-deploy/entrypoint.sh [hunk 4]
```diff
@@ -2,80 +2,90 @@
...
+    cd $AZURE_APP_PACKAGE_LOCATION
+    zip -r "$NEW_PACKAGE_LOCATION" * > /dev/null
+    AZURE_APP_PACKAGE_LOCATION="$NEW_PACKAGE_LOCATION"
+    cd "$GITHUB_WORKSPACE"
+    echo "Compressed package. New Package path: '${AZURE_APP_PACKAGE_LOCATION}'"
+fi
+
+APP_KIND=`az resource show -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" --resource-type "Microsoft.Web/Sites" --query 'kind'` 
+
 echo "Web App type : ${APP_KIND}"
```
Comment (SumiranAgg) [inline function-app-deploy/entrypoint.sh]: function app

File: function-app-deploy/entrypoint.sh [hunk 5]
```diff
@@ -2,80 +2,90 @@
...
 
 echo "Initiating Web App Deployment"
+
 if [[ ! $APP_KIND =~ $LINUX_APP_SUBSTRING ]];
```
Comment (SumiranAgg) [inline function-app-deploy/entrypoint.sh]: Variable not initialized LINUX_APP_SUBSTRING

File: web-app-deploy/entrypoint.sh [hunk 6]
```diff
@@ -2,74 +2,81 @@
...
+fi
+
+APP_KIND=`az resource show -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" --resource-type "Microsoft.Web/Sites" --query 'kind'` 
+
 echo "Web App type : ${APP_KIND}"
 
 echo "Initiating Web App Deployment"
-if [[ ! $APP_KIND =~ $LINUX_APP_SUBSTRING ]];
+
+if [[ ! $APP_KIND =~ "linux" ]];
```
Comment (SumiranAgg) [inline web-app-deploy/entrypoint.sh]: do you want to use string or variable? Also if variable can we define it at the top of the script as we do will constants for better visibility during future changes

Comment (vincentdass) [reply web-app-deploy/entrypoint.sh]: yes, created variable

File: function-app-deploy/entrypoint.sh (hunk 5 above)
Comment (vincentdass) [reply function-app-deploy/entrypoint.sh]: updated. thanks

File: function-app-deploy/entrypoint.sh (hunk 3 above)
Comment (vincentdass) [reply function-app-deploy/entrypoint.sh]: updated.

File: function-app-deploy/entrypoint.sh (hunk 4 above)
Comment (vincentdass) [reply function-app-deploy/entrypoint.sh]: updated.

**Review by SumiranAgg — APPROVED** [review]

**Pull request merged**